### DISCUSSION TITLE
How BlockSuite not enable the contenteditable container yet still utilize browser selection with cross-text selection?

### DISCUSSION THREAD

**Discussion by JeffAAA** [toeverything/blocksuite#4217] [Q&A] [answered]
Since BlockSuite does not use a single contenteditable container to enable cross-text selection
![image](https://github.com/toeverything/blocksuite/assets/20382725/dea9a996-c783-4845-8900-537a9f72d883)

I though you guys might be drawing your own selection. However, when I changed the browser's selection background, it turned out that BlockSuite still use the browser's selection.

This situation is quite peculiar. How can BlockSuite not enable the contenteditable container yet still utilize browser selection with cross-text selection?

I'm confused.

you can try this [code pen](https://codepen.io/kmdxzdgt-the-sasster/pen/OJBYwVB), we can not drag cross text selection without a contentediable container.
![2023-08-18 16 34 08](https://github.com/toeverything/blocksuite/assets/20382725/5bfb8824-ff26-43ab-8711-8625e775cee9)

**Comment by doodlewind** [comment, marked as answer]
Hi, thanks for your interest in BlockSuite! The way we support cross-contenteditable selection is well explained in https://blocksuite.affine.pro/rich-text-orchestration.html#flat-range-selection, hope this helps!

**Reply by JeffAAA** [reply]
Thanks for reply!

I realized that BlockSuite uses mouse coordinates to convert them into browser ranges, and then utilizes that information to update the browser selection. This approach works effectively in scenarios involving drag selection.

However, when I attempted some other scenarios involving cross-text selection, I found that we cannot fully leverage the browser's selection functionality for such cases. For instance, scenarios like dragging to perform cross-text selection on a mobile device or using the shift key and directional keys to achieve cross-text selection may not be fully compatible.

If we intend to employ cross-text selection within a non-editable container, it might be more suitable to option for a self-drawn selection approach.

I appreciate your response to my question!
